src/ContractionTreeToGraph.jl

```diff
@@ -0,0 +1,120 @@
+            for vn in setdiff(vertices(g), [v])
+                vn_set = [Set(vni) for vni in vn]
+                if(Set(concat1) ∈ vn_set || Set(concat2) ∈ vn_set || Set(concat3) ∈ vn_set)
+                    add_edge!(g, v => vn)
+                end
+            end
+        end
+    end
+
+
+    return g
+end
+
+"""Given a contraction sequence which is a subsequence of some larger sequence which is being built on current_g and has leaves `leaves`
+Spawn `contract sequence' as a vertex on `current_g' and continue on with its children """
+function spawn_child_branch(contract_sequence, current_g, leaves)
+    #Check if sequence is at leaf point
+    if(isa(contract_sequence, Array))
+        #If not it  will spawn two children
+        group1 = collect(Leaves(contract_sequence[1]))
+        group2 = collect(Leaves(contract_sequence[2]))
+        remaining_verts = setdiff(leaves, vcat(group1, group2))
+        add_vertex!(current_g, (group1, group2, remaining_verts))
+        spawn_child_branch(contract_sequence[1], current_g, leaves)
+        spawn_child_branch(contract_sequence[2], current_g, leaves)
+    else
+        #If it is it is just a vertex
+        add_vertex!(current_g, ([contract_sequence], setdiff(leaves, [contract_sequence])))
+    end
+end
+
+"""Utility functions for the graphical representation of a contraction sequence
+Perhaps it should be a specific object or type of tree?!"""
+
+"""Determine if a node is a leaf"""
+function leaf_node(g::AbstractGraph, v)
+    return length(neighbors(g, v)) == 1
+end
+
+"""Determine if an edge involves a leaf (at src or dst)"""
+function leaf_edge(g::AbstractGraph, e)
+    return leaf_node(g, src(e)) || leaf_node(g, dst(e))
+end 
+
+"""Determine if a node has no neighbours which are leaves"""
+function no_leaf_neighbours(g::AbstractGraph, v)
+    for vn in neighbors(g, v)
+        if(leaf_node(g, vn))
+            return false
+        end
+    end
+    return true
+end
+
+"""Get all edges which do not involve a leaf
+(why does edges(g)[contraction_edge(g, edges(g)) .== true] not work?)"""
+function non_leaf_edges(g::AbstractGraph)
+    return edges(g)[findall(==(1), [!leaf_edge(g,e) for e in edges(g)])]
+end
+
+"""Get all nodes which aren't leaves and don't have leaf neighbours (these represent internal points in the contraction sequence)"""
+function internal_nodes(g::AbstractGraph)
+    return vertices(g)[findall(==(1), [no_leaf_neighbours(g, v) && !leaf_node(g, v) for v in vertices(g)])]
+end
+
+"""Get all nodes which have leaf neighbours but aren't leaves themselves (these represent start/ end points of a contraction sequence)"""
+function external_nodes(g::AbstractGraph)
+    return vertices(g)[findall(==(1), [!no_leaf_neighbours(g, v) && !leaf_node(g,v)  for v in vertices(g)])]
+end
+
+"""Get the vertex bi-partition that a given edge between non-leaf nodes represents"""
+function edge_bipartition(g::AbstractGraph, e)
```

Comment:
So this function is a bit tricky to name since it isn't really for generic tree graph, since it assumes information about the leaves are stored within the vertices.

To clarify that, I would call this `contraction_tree_leaf_bipartition` (where in principle there could be a `leaf_bipartition` function that just analyzed the graph itself but did the same thing, which would work for any tree graph).

It hits a bit on a question about terminology. So far we've used `contraction_sequence` as the terminology for the nested vector data structure, so it would be good to have a consistent terminology for the graph representation. We could just go with `contraction_tree` for that, which implies it is a tree graph.